src/main.rs

```diff
@@ -182,6 +182,88 @@ fn new_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
 }
 
 
+/// reply subcommand implementation
+///
+fn reply_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    let sig = try_or_1!(repo.signature());
+
+    // NOTE: We want to do a lot of stuff early, because we want to report
+    //       errors before a user spent time writing a commit message in her
+    //       editor. This means that we have a lot of bindings which may not
+    //       be neccessary otherwise, resulting in data lying around.
+
+    // the unwrap is safe since `parent` is a required value
+    // and get all the info from it that we might need
+    let mut parent = try_or_1!(repo.value_to_commit(matches.value_of("parent").unwrap()));
+
+    // extract the subject and tree from the parent
+    let subject = parent.summary().map(|s| {
+        if s.starts_with("Re: ") {
+            s.to_owned()
+        } else {
+            format!("Re: {}", s)
+        }
+    });
+    let tree = try_or_1!(parent.tree());
+
+    // figure out to what issue we reply
+    // NOTE: The following is this ugly because `Clone` is not implemented for
+    //       `git2::Commit`, `Debug` is not implemented for `git2::Object` and
+    //       `find_tree_init()` consumes the commit because of internals.
+    let issue = try_or_1!(repo.find_tree_init(parent.as_object().clone().into_commit().ok().unwrap())).id();
```

Comment:
Maybe it makes sense to have `find_tree_init()` take a reference (and move the ugly workaround over there). It would clearly make sense regarding semantics. The sometimes unnecessary clone would be a drawback, but this operation _should_ be relatively cheap.

I'm open to suggestions here.